src/shared/settingsholder.cpp

```diff
@@ -296,6 +297,8 @@ bool SettingsHolder::rollbackTransaction() {
     QMetaObject::invokeMethod(this, i.value().first, Qt::DirectConnection);
   }
 
+  emit transactionRolledBack();
```

Comment:
We should emit the signal even if `finalizeTransition` returns false to avoid inconsistent states.